mod_ci/controllers.py

```diff
@@ -341,6 +342,61 @@ def mark_test_failed(db, test, repository, message: str) -> None:
         log.error(f"Failed to mark test {test.id} as failed: {e}")
 
 
+def _verify_binary_commit(binary_path: str, expected_commit: str, log) -> Tuple[bool, str]:
+    """
+    Verify that the binary's embedded commit SHA matches the expected commit.
+
+    Runs the binary with --version and parses the output to extract the Git commit.
+    This ensures we're testing the correct binary and prevents race conditions.
+
+    :param binary_path: Path to the ccextractor binary
+    :type binary_path: str
+    :param expected_commit: The expected Git commit SHA
+    :type expected_commit: str
+    :param log: Logger instance
+    :return: Tuple of (success, message)
+    :rtype: tuple[bool, str]
+    """
+    try:
+        result = subprocess.run(
+            [binary_path, '--version'],
```

Comment:
We do not want to run this on our platform; if the binary is malicious this can infect everything.